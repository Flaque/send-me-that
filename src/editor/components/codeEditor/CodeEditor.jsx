import React from 'react';
import CodeMirror from 'react-codemirror';
import 'codemirror/mode/javascript/javascript';

class CodeEditor extends React.Component {

  updateCode(newCode) {
    this.setState({
      code: newCode
    })
    this.props.onCodeChange(newCode);
  }

  constructor(props) {
    super(props)
    this.state = { code: this.props.initialCode }
    this.updateCode = this.updateCode.bind(this)

    this.options = {
      lineNumbers: true,
      mode: 'javascript'
    }
  }

  render() {
    return (
      <CodeMirror
      className='bordered'
      value={this.state.code}
      onChange={this.updateCode}
      options={this.options} />
    )
  }
}

export default CodeEditor;
