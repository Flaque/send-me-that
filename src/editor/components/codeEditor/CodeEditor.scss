$editor-max-height: 300px;

.app {
  .ReactCodeMirror {
    border-top-right-radius: 3px;
    border-top-left-radius: 3px;
    max-height: $editor-max-height;
    display: flex;

    .CodeMirror-gutters {
      border-right: 1px solid #eee;
    }
  }

  .CodeMirror {
    border: 1px solid #eee;
    height: auto;
    width: 100%;

    .CodeMirror-scroll {
      max-height: $editor-max-height;
    }
  }
}
