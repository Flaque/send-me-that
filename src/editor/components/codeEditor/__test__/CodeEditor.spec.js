import React from 'react';
import CodeEditor from '../CodeEditor.jsx';
import renderer from 'react-test-renderer';

test('Test that CodeEditor is still the same', () => {
  const tree = renderer.create(
    <CodeEditor />
  ).toJSON();
  expect(tree).toMatchSnapshot();
})
