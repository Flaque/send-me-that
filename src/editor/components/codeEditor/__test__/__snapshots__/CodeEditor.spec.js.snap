exports[`test Test that CodeEditor is still the same 1`] = `
<CodeMirror
  className="bordered"
  onChange={[Function]}
  options={
    Object {
      "lineNumbers": true,
      "mode": "javascript",
    }
  }
  preserveScrollPosition={false}
  value={undefined} />
`;
